src/instrumentations/extentions/aws/SQSEventCreator.ts

```diff
@@ -0,0 +1,128 @@
+        // span.setAttribute(
+        //   'aws.sqs.record.message_attributes',
+        //   JSON.stringify(requestInfo.request.commandInput.MessageAttributes)
+        // );
+        span.setAttribute(
+          'aws.sqs.record.delay_seconds',
+          requestInfo.request.commandInput.DelaySeconds
+        );
+        break;
+      case 'SendMessageBatch':
+        span.setAttribute(
+          'aws.sqs.entries',
+          JSON.stringify(requestInfo.request.commandInput.Entries)
+        );
+        break;
+      case 'ReceiveMessage':
+        // span.setAttribute(
+        //   'aws.sqs.record.message_attributes',
+        //   requestInfo.request.commandInput.MessageAttributeNames
+        // );
+        span.setAttribute(
+          'aws.sqs.visibility_timeout',
+          requestInfo.request.commandInput.VisibilityTimeout
+        );
+        span.setAttribute(
+          'aws.sqs.wait_time_seconds',
+          requestInfo.request.commandInput.WaitTimeSeconds
+        );
+        span.setAttribute(
+          'aws.sqs.max_number_of_messages',
+          requestInfo.request.commandInput.MaxNumberOfMessages
+        );
+        break;
+    }
+  }
+  responseHandler(
+    span: Span,
+    responseInfo: AwsSdkResponseHookInformation
+  ): void {
+    switch (responseInfo.response.request.commandName) {
+      case 'SendMessage':
+        span.setAttribute(
+          'aws.sqs.record.message_id',
+          responseInfo.response.data.MessageId
+        );
+        break;
+      case 'SendMessageBatch':
+        span.setAttribute(
+          'aws.sqs.record',
+          JSON.stringify(responseInfo.response.data)
+        );
+        break;
+      case 'ReceiveMessage': {
+        if (responseInfo.response.data.Messages.length == 1) {
+          span.setAttribute(
+            'aws.sqs.record.message_body',
+            responseInfo.response.data.Messages[0].Body
+          );
+          span.setAttribute(
+            'aws.sqs.record.message_attributes',
+            JSON.stringify(
+              responseInfo.response.data.Messages[0].MessageAttributes
+            )
+          );
+          span.setAttribute(
+            'aws.sqs.record.message_id',
+            responseInfo.response.data.Messages[0].MessageId
+          );
+          span.setAttribute(
+            'aws.sqs.record.attributes.sender_id',
+            responseInfo.response.data.Messages[0].Attributes.SenderId
+          );
+          span.setAttribute(
+            'aws.sqs.record.attributes.sent_timestamp',
+            responseInfo.response.data.Messages[0].Attributes.SentTimestamp
+          );
+          span.setAttribute(
+            'aws.sqs.record.attributes.aws_trace_header',
+            responseInfo.response.data.Messages[0].Attributes.AWSTraceHeader
+          );
+        }
+        if (responseInfo.response.data.Messages.length > 1) {
+          span.setAttribute(
+            'aws.sqs.record',
+            JSON.stringify(responseInfo.response.data.Messages)
```

Comment:
same https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md.

The attributes labels should be flat

Reply:
This is the case of receiving multiple messages with the same attributes. We should think how to declare it in the spec